### ISSUE TITLE
Fetch surrounding chunks notebook

### ISSUE THREAD

**Issue by sunilemanjee** [elastic/elasticsearch-labs#257]
I have built a notebook to demonstrate how to fetch surrounding chunks within Elasticsearch. This is not the only way to do it, but it is definitely a valid approach.

Notebook: [Google Colab](https://colab.research.google.com/drive/1RX97YPGdOaNxCg3vnceCVYluk24kDOCo?usp=sharing#scrollTo=xJBDwRmDQq4n)

The example uses text from a Harry Potter book, splitting it by chapter and then into chunks. Each chunk is a nested passage containing the text, dense, and sparse representations of the chunk.

When searching, the demo will fetch the matching passage chunk along with surrounding chunks. If the chunk is the first chunk in the chapter, it will fetch n, n+1, and n+2. If it is the last chunk, it will fetch n, n-1, and n-2.

**Comment by joemcelroy** [comment]
Hey Sunile - Worth creating a search labs article on this. i would be hesitant that it should be something we advise and in future, we hope that semantic_text will have this option of expanding hit passages.

**Comment by sunilemanjee** [comment]
I agree on publishing an article. My goal is to publish the notebook and
then shortly after an article.

WRT your hesitation on advising this approach, mind elaborating on this?
Fetching surrounding chunks is a pattern for RAG applications.


Regards,

Sunile Manjee | Senior Principal Solutions Architect | 630-333-2433
*Azure Solutions Architect Expert Certified, Accreditation verified here
<https://tinyurl.com/3mtzu33t>*
*Azure Fundamentals Certified, Accreditation verified here
<https://tinyurl.com/bdte7p7v>*
Technical Enablement: ***@***.***
<https://tinyurl.com/ywxxt8te>

<https://www.elastic.co/>

**Comment by joemcelroy** [comment]
Oh im fine with advising this approach given the current situation today, but makes sense in the supporting-blog-content and more giving visibility that they can do this. When we have this baked into Elasticsearch, we can add it into notebooks which is a more formalised and supported way.

`/supporting-blog-content` = Search labs article + notebook example
`/notebooks` = product feature of elasticsearch thats well supported
